mmv1/products/discoveryengine/ChatEngine.yaml

```
@@ -0,0 +1,163 @@
+
+--- !ruby/object:Api::Resource
+name: 'ChatEngine'
+description: |
+  Vertex chat and Conversation Engine Chat type
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Vertex AI Conversation': 'https://cloud.google.com/dialogflow/cx/docs/concept/generative'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+update_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+autogen_async: true
+
+async: !ruby/object:Api::OpAsync
+  actions: ['create', 'delete']
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+  result: !ruby/object:Api::OpAsync::Result
+    resource_inside_response: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/discovery_engine_chat_engine_hardcode_solution_type.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engineId'
+    description: |
+      The ID to use for chat engine.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the chat engine. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/engines/{engine_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
+    description: |
+      The industry vertical that the chat engine registers.
+    values:
+      - :GENERIC
+    immutable: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      displayName
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    min_size: 1
+    description: |
+      Data Store IDs, currently modification of Data Stores after creation is not supported.
```

```suggestion
      The data stores associated with this engine. Multiple DataStores in the same Collection can be associated here. All listed DataStores must be `SOLUTION_TYPE_CHAT`. Adding or removing data stores will force recreation.
```

Maybe something like this?